Source/Scene/Model/I3dmLoader.js

```diff
@@ -635,14 +635,62 @@ function createInstances(loader, components, frameState) {
   instances.featureIds.push(featureIdInstanceAttribute);
 
   // Apply instancing to every node that has at least one primitive.
-  for (i = 0; i < components.nodes.length; i++) {
+  const nodes = components.nodes;
+  const nodesLength = nodes.length;
+  let makeInstancesCopy = false;
+  for (i = 0; i < nodesLength; i++) {
     const node = components.nodes[i];
     if (node.primitives.length > 0) {
-      node.instances = instances;
+      // If the instances have not been assigned to a node already, assign
+      // it to the first node encountered. Otherwise, make a copy of them
+      // for each subsequent node.
+      node.instances = makeInstancesCopy
+        ? createInstancesCopy(instances)
+        : instances;
+
+      makeInstancesCopy = true;
     }
   }
 }
 
+/**
+ * Returns a copy of the instances that contains shallow copies of the instanced
+ * attributes. That is, the instances and attribute objects will be new copies,
+ * but they will point to the same buffers and typed arrays. This is so each
+ * node can manage memory separately, such that unloading memory for one
+ * node does not unload it for another.
+ *
+ * @returns {ModelComponents.Instances}
+ *
+ * @private
+ */
+function createInstancesCopy(instances) {
+  const instancesCopy = new Instances();
+  instancesCopy.transformInWorldSpace = instances.transformInWorldSpace;
+
+  const attributes = instances.attributes;
+  const attributesLength = attributes.length;
+
+  for (let i = 0; i < attributesLength; i++) {
+    const attribute = attributes[i];
+    const attributeCopy = new Attribute();
+    attributeCopy.name = attribute.name;
+    attributeCopy.setIndex = attribute.setIndex;
+    attributeCopy.semantic = attribute.semantic;
+    attributeCopy.componentDatatype = attribute.componentDatatype;
+    attributeCopy.type = attribute.type;
+    attributeCopy.count = attribute.count;
+    attributeCopy.typedArray = attribute.typedArray;
+    attributeCopy.buffer = attribute.buffer;
```

Comment:
Can cloning the individual attribute be done like this?

```
const attributeCopy = clone(attribute, false);
```

It would simplify the code and make it more easy to maintain.